repo/blob/s3/s3_storage_test.go

```diff
@@ -194,10 +194,21 @@ func TestS3StorageAWSRetentionUnversionedBucket(t *testing.T) {
 		Region:          getEnvOrSkip(t, testRegionEnv),
 	}
 
-	createBucket(t, options)
-	testPutBlobWithInvalidRetention(t, options, blob.PutOptions{
-		RetentionMode:   minio.Governance.String(),
-		RetentionPeriod: time.Hour * 24,
+	getOrCreateBucket(t, options)
+
+	t.Run("valid period", func(t *testing.T) {
+		// expected to fail on non-locked buckets
+		testPutBlobWithInvalidRetention(t, options, blob.PutOptions{
+			RetentionMode:   minio.Governance.String(),
+			RetentionPeriod: time.Hour * 24,
+		})
+	})
+
+	t.Run("invalid period", func(t *testing.T) {
```

Comment:
@Shrekster this test fails for me. That is the put blob succeeds even when the retention period is a nanosecond.